### PR TITLE
CI: update mergify rules

### LINKED ISSUE #1016: Update Mergify rules for commitlint bot and DCO bot
currently, Mergify will merge the PR if the Travis CI is a success. we need to enhance the Mergify rules to merge the PR if all the 3 rules are met.

### PR DESCRIPTION
**Pull request by Madhu-1**
Updated mergify rules to consider DCO and commitlint status

Fixes #1016

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -14,6 +14,8 @@ pull_request_rules:
       - "#approved-reviews-by>=2"
       - "#changes-requested-reviews-by=0"
       - "status-success=continuous-integration/travis-ci/pr"
+      - "status-success=DCO"
+      - "status-success=commitlint"
```
Comment (nixpanic) [inline .mergify.yml:18]: What does `commitlint` check? Can you link to a description in the mergify docs?

Comment (Madhu-1) [reply .mergify.yml:18]: commitlint is a bot added in cephcsi repo and the rules are valid, you can check the rules are valid or not here https://github.com/ceph/ceph-csi/pull/1017/checks?check_run_id=648486513

Comment (nixpanic) [reply .mergify.yml:18]: Well, yes, I understand that part. But what kind of requirements does `commitlint` have on the commit messages?

Comment (Madhu-1) [reply .mergify.yml:18]: https://github.com/ceph/ceph-csi/pull/1015 adds the required details on how the commit message should be like

Comment (nixpanic) [reply .mergify.yml:18]: Right, this should probably all have been done in the same PR :man_shrugging:

**Comment by Madhu-1** [comment]
Merging as it needs manual merge

**Pull request merged**